manifests/backup/xtrabackup.pp

```diff
@@ -1,8 +1,8 @@
 # See README.me for usage.
 class mysql::backup::xtrabackup (
   $xtrabackup_package_name = $mysql::params::xtrabackup_package_name,
-  $backupuser              = '',
-  $backuppassword          = '',
+  $backupuser              = undef,
+  $backuppassword          = undef,
```

Comment:
It seems like this might be a good chance to start using types and label these as `Optional[String]`.